crates/db_views/src/post_view.rs

```diff
@@ -185,6 +186,34 @@ fn queries<'a>() -> Queries<
     };
 
   let list = move |mut conn: DbConn<'a>, options: PostQuery<'a>| async move {
+    macro_rules! order_and_page_filter_desc {
```

Comment:
i've tried to do this without a macro and without copy-pasting this code 10 times, but I couldn't figure out the types for diesel if this was a lambda or function. I'm not sure if it would even be possible, even with BoxedQuery..

Reply:
I'd almost rather this be copy-pasted around, if even some parts of it could be extracted into functions, rather than using macros. @dullbananas might be able to help with diesel function return types.

Reply:
The type of `query` should be inferrable if a closure is used.

The type of the column is tricky because columns with multiple different sql types are used. If only one sql types was needed, then `BoxableExpression` could be used.

Reply:
I tried using a closure, but even then you need to be able to name the return type and I couldn't manage to figure that out..

Reply:
A function like this could probably be used

```
fn order_and_page_filter_desc<Q, C, T>(
  query: &mut Q,
  column: C,
  options: &PostQuery,
  getter: impl Fn(&PostAggregates) -> T,
)
where
  Q: ThenOrderDsl<dsl::Desc<C>, Output = Q> + FilterDsl<dsl::GtEq<C, T>, Output = Q> + FilterDsl<dsl::GtEq<C, T>, Output = Q>,
  C: Expression,
  C::SqlType: SingleValue + SqlType,
  T: AsExpression<C::SqlType>,
```

Usage:

```
order_and_page_filter_desc(&mut query, post_aggregates::featured_local, &options, |a| a.featured_local);
```

The ascending version would use `dsl::Asc` instead of `dsl::Desc`

Reply:
`*query = query.then_order_by(column.desc());` seems to not work because of `cannot move out of `*query` which is behind a mutable reference
move occurs because `*query` has type `Q`, which does not implement the `Copy` trait

Reply:
Remove the `&mut` and make the function return the same type used for the `query` parameter